pkg/irmacontract.go

```diff
@@ -133,3 +168,42 @@ func validateActingParty(params map[string]string, actingParty string) (bool, er
 	// perform the actual check
 	return actingParty == actingPartyFromContract, nil
 }
+
+func (cv *IrmaContractVerifier) verifyRequiredAttributes(signedIrmaContract *SignedIrmaContract, validationResult *ContractValidationResult) (*ContractValidationResult, error) {
+	if validationResult.ValidationResult == Invalid {
+		return validationResult, nil
+	}
+
+	contractTemplate := signedIrmaContract.ContractTemplate
+
+	// use a map to ignore duplicates. Allows us to compare lengths
+	validationRes := make(map[string]bool)
+
+	requiredAttributes := contractTemplate.SignerAttributes
+
+	for disclosedAtt := range validationResult.DisclosedAttributes {
+		// e.g. gemeente.personalData.firstnames
+		for _, requiredAttribute := range requiredAttributes {
+			// e.g. .gemeente.personalData.firstnames
+			if strings.HasSuffix(requiredAttribute, disclosedAtt) {
+				validationRes[requiredAttribute] = true
+			}
+		}
+	}
+
+	if len(validationRes) != len(requiredAttributes) {
+		foundAttributes := make([]string, len(validationRes))
+		for k := range validationRes {
+			foundAttributes = append(foundAttributes, k)
+		}
+
+		disclosedAttributes := make([]string, len(validationResult.DisclosedAttributes))
```

Comment:
You're just copying the string array? That can be done shorter:

`disclosedAttributes := append([]string(nil), validationResult.DisclosedAttributes...)`

Reply:
I'm copying the keys of the map.

Reply:
I failed to notice that :+1: